### PR TITLE
Add/serve JSON data for toast nofitications

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

- Added json file for toast notifications
- Add route from app.py. I'm not sure how to serve all `.json` files in the directory, so for the time being have just served the toast notifications file

Fixes [#1203](https://github.com/canonical-web-and-design/vanilla-squad/issues/1203)

## QA

- Open [demo](https://vanilla-framework-4315.demos.haus/)
- go to https://vanilla-framework-4315.demos.haus/docs/examples/patterns/notifications/toast.json and check the json data exists

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [ ] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

File: webapp/app.py [hunk 1]
```diff
@@ -163,6 +163,13 @@ def examples_index():
     )
 
 
+@app.route("/docs/examples/patterns/notifications/toast.json")
```
Comment (bartaz) [inline webapp/app.py:166]: I don't know exactly either, but something like that could work?

```
@app.route("/<path:json_path>.json")
def json_examples(json_path):
```

And then you could use `json_path` to retrieve the file (I guess you would need to add `.json` to it for full file name.

Try to check some other examples where part of URL is passed to the function like in line 180:

```
@app.route("/docs/examples/standalone/<path:example_path>")
def standalone_example(example_path):
```

Comment (carkod) [reply webapp/app.py:166]: But if you load them dynamically in the same endpoint, are all the json files going to be in the same folder? Wouldn't it make sense to separate them like they are now?

Comment (mrgnr) [reply webapp/app.py:166]: The following works for me:

```python
@app.route("/<path:json_path>.json")
def json_examples(json_path):
    try:
        return flask.send_file(f"{app.template_folder}/{json_path}.json")
    except FileNotFoundError:
        return flask.abort(404)
```

**Review by mrgnr — APPROVED** [review]
LGTM :+1:

**Pull request merged**